transports/tcp/src/lib.rs

```diff
@@ -531,9 +417,10 @@ where
         if !is_tcp_addr(listen) || !is_tcp_addr(observed) {
             return None;
         }
-        match &self.port_reuse {
-            PortReuse::Disabled => address_translation(listen, observed),
-            PortReuse::Enabled { .. } => Some(observed.clone()),
+        if self.port_reuse.already_dialed_as_listener(&observed) {
+            Some(observed.clone())
+        } else {
+            address_translation(listen, observed)
         }
     }
```

Comment:
Prefer early-return over `else`.

```suggestion
        if self.port_reuse.already_dialed_as_listener(&observed) {
            return Some(observed.clone())
        }
        
        address_translation(listen, observed)
    }
```

Reply:
Again a question of preferred style, but I will change it.